cmd/replace.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"regexp"
+
+	"github.com/spf13/cobra"
+
+	"github.com/hacbs-contract/ec-cli/internal/image"
+)
+
+type replaceFn func([]string, string) ([]byte, error)
+
+func replaceCmd(r replaceFn) *cobra.Command {
+	var data = struct {
+		Images     []string
+		Source     string
+		Replace    bool
+		OutputFile string
+	}{}
+
+	cmd := &cobra.Command{
+		Use:   "replace",
+		Short: "Replace image references in the given input",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+
+			out, err := r(data.Images, data.Source)
+			if err != nil {
+				return err
+			}
+
+			if data.OutputFile == "" {
+				fmt.Println(string(out))
+			} else {
+				f, err := os.Create(data.OutputFile)
+				if err != nil {
+					return err
+				}
+				defer f.Close()
+				_, err = f.Write(out)
+				if err != nil {
+					return err
+				}
+			}
+
+			if data.Replace {
+				stat, err := os.Stat(data.Source)
+				if err != nil {
+					return err
+				}
+				f, err := os.OpenFile(data.Source, os.O_RDWR, stat.Mode())
+				if err != nil {
+					return err
+				}
+				defer f.Close()
+				_, err = f.Write(out)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&data.Source, "source", "s", data.Source,
+		"REQUIRED - An existing YAML file")
+
+	cmd.Flags().StringSliceVarP(&data.Images, "image", "i", data.Images,
+		"REQUIRED - The bundle image reference to  track - may be used multiple times")
+
+	cmd.Flags().BoolVarP(&data.Replace, "replace", "r", data.Replace,
+		"Ovewrite source file with changes.")
+
+	cmd.Flags().StringVarP(&data.OutputFile, "output", "o", data.OutputFile,
+		"Write changes to a file. Use empty string for stdout, default behavior")
+
+	// TODO: We should check the error result here
+	_ = cmd.MarkFlagRequired("image")
+	_ = cmd.MarkFlagRequired("source")
+
+	return cmd
+}
+
+func init() {
+	r := replaceCmd(replace)
+	rootCmd.AddCommand(r)
+}
+
+type ImageReplacer struct {
+	*image.ImageReference
+	regex *regexp.Regexp
+}
+
+func (i *ImageReplacer) Match(b []byte) bool {
+	return i.regex.Match(b)
+}
+
+func (i *ImageReplacer) Replace(b []byte) []byte {
```

Comment:
Probably doesn't need to be public

```suggestion
func (i *ImageReplacer) replace(b []byte) []byte {
```